src/main/java/bio/terra/cli/command/resource/addref/GcsObject.java

```diff
@@ -41,10 +52,29 @@ protected void execute() {
     // build the resource object to add
     CreateResourceParams.Builder createResourceParams =
         referencedResourceCreationOptions.populateMetadataFields();
+
+    // parsing the path as bucket name and object name
+    if (gcsPath != null) {
+      if (bucketName != null || objectName != null) {
+        throw new UserActionableException("Specify only one path to add reference.");
+      } else {
+        Pattern r = Pattern.compile("(?:^gs://)([^/]*)/(.*)");
+        Matcher m = r.matcher(gcsPath);
+        if (m.find()) {
+          bucketName = m.group(1);
+          objectName = m.group(2);
+        }
+      }
+    } else {
+      if (bucketName == null || objectName == null) {
```

Comment:
move this to top and throw early. then we get rid of some nested if statements.

also update the error. something like both bucketName and objectName is required.